csharp/src/Ice/Network.cs

```diff
@@ -217,28 +217,7 @@ internal static Socket CreateSocket(bool udp, AddressFamily family)
             bool preferIPv6,
             CancellationToken cancel)
         {
-            // For client endpoints, an empty host is the same as the loopback address
-            if (host.Length == 0)
-            {
-                var addresses = new List<IPEndPoint>();
-                foreach (IPAddress a in GetLoopbackAddresses(ipVersion))
-                {
-                    addresses.Add(new IPEndPoint(a, port));
-                }
-
-                if (ipVersion == EnableBoth)
-                {
-                    if (preferIPv6)
-                    {
-                        return addresses.OrderByDescending(addr => addr.AddressFamily);
-                    }
-                    else
-                    {
-                        return addresses.OrderBy(addr => addr.AddressFamily);
-                    }
-                }
-                return addresses;
-            }
+            Debug.Assert(host.Length > 0);
```

Comment:
Wouldn't this result in an Assert when you receive a endpoint without a Host?

A 3.7 client can marshal an endpoint without a Host an will trigger this assert, or a malicious client can marshal it by hand and bring the server down.

If we want to keep this assert we should either throw when unmarshal an Endpoint without a Host or set the Host to a default valid value.

Reply:
https://github.com/zeroc-ice/ice/blob/0007b7bb4d66b523bb94825f64d568df94905045/csharp/src/Ice/IPEndpoint.cs#L281

We don't allow an empty host anymore. I'm pretty sure Ice 3.7 (not misbehaving) will always marshal the host of an endpoint.

Reply:
You are right 3.7 will marshal the default host, still a malicious client can trigger that by manually marshalling the endpoint without a host, I think the IPEndpoint unmarshalling constructor should throw `InvalidDataException` in this case.

Reply:
Good catch, I missed the other constructor.